src/timely-util/src/builder_async.rs

```diff
@@ -130,9 +136,31 @@ impl<'handle, T: Timestamp, D: Container, P: Pull<BundleCore<T, D>>> Future
                 // None branch is replaced by a panic!(). Therefore each individual path is safe on
                 // its own and in the full version we only ever call one of them, therefore the
                 // composition is also safe.
-                let cap =
-                    unsafe { transmute::<CapabilityRef<'_, T>, CapabilityRef<'handle, T>>(cap) };
+                //
+                // ## Safety of ManuallyDrop
+                // `next_sync` wraps the returned `CapabilityRef` into a `ManuallyDrop` in order to
+                // not interfere with the borrow checker that will otherwise infer that `cap` may
+                // live until the end of this function and doesn't compile even when transmuting.
```

Comment:
```suggestion
                // `next_sync` wraps the returned `CapabilityRef` into a `ManuallyDrop` in order to
                // not interfere with the borrow checker that will otherwise infer that `cap` may
                // live until the end of this function and doesn't compile even when transmuting.
                // Its unclear why it infers this
```